packages/nimble-components/src/banner/specs/README.md

```diff
@@ -0,0 +1,186 @@
+- *Methods*
+- *Events*
+  - `close` - fired when the banner is dismissed
+  - `action` - fired when the action button is activated (by mouse or keyboard). This is the intended way for clients to perform an action.
+- *CSS Classes and CSS Custom Properties that affect the component*
+
+### Alternatives
+
+- The text of the banner could be provided as content rather than an attribute. This has the benefit of leaving the door open for us to support more than just plaintext in the future. However, so long as we _don't_ want to support arbitrary HTML, it may have the downside of giving users the wrong idea.
+
+- Instead of configuring the action link/button via attributes, we could allow the user to provide that element as content and slot it in an `action` slot. This would have the following benefits:
+    - allows full configuration of action element (e.g. attributes like `target` on a `nimble-anchor`)
+    - user can add handlers directly to their action element rather than relying on `action` event
+    - easier to support multiple actions in the future
+    - easy to support other kinds of action elements in the future
+
+    However, that flexibility is also a downside. We would no longer have tight control over what the user could show and how it would look.
+
+### Anatomy
+
+```html
+<div class="icon">
+  ${when(x => x.type === 'error', html`<nimble-icon-exclamation-mark></nimble-icon-exclamation-mark>`)}
+  ${when(x => x.type === 'warning', html`<nimble-icon-triangle-filled></nimble-icon-triangle-filled>`)}
+  ${when(x => x.type === 'info', html`<nimble-icon-info></nimble-icon-info>`)}
+</div>
+<div class="text">
+  <span class="heading">
+    ${x => x.heading}
+  </span>
+  ${x => x.text}
+</div>
+${when(x => x.actionText, html`
+  <div class="action">
+    ${when(x.actionHref, html`
+      <nimble-anchor href="${x => x.actionHref}">${x => x.actionText}</nimble-anchor>`
+    )}
+    ${when(!x.actionHref, html`
+      <nimble-button appearance="${x => x.actionButtonAppearance}">${x => x.actionText}</nimble-button>`
+    )}
+  </div>`
+)}
+<div class="close">
+  ${when(x => !x.preventDismiss), html`
+    <nimble-button appearance="ghost" content-hidden>
+      <nimble-icon-xmark slot="start"></nimble-icon-xmark>
+    </nimble-button>`
+  )}
+</div>
+```
+
+### Angular integration
+
+An Angular wrapper will be created for `nimble-banner`. No `ControlValueAccessor` is needed.
+
+`routerLink` will not be supported for the link.
+
+### Blazor integration
+
+Blazor integration for the `nimble-banner` will be provided.
+
+### Visual Appearance
+
+See XD document link at the top of this document.
+
+The biggest issue is that we will have to completely re-style the icons, link, and buttons:
+- Primary icon (error, warning, or info) color is transparent white (in all themes)
+- Action button height is 24px (instead of 32px)
+- Action button hover border color is white (in all themes)
```

Comment:
Can some of these color issues be resolved by internally wrapping some components in a theme provider element that is configured to 'color' theme?

Reply:
It might resolve some of those issues by coincidence, but that doesn't seem like the right solution since we're not actually trying to use the color theme. There are variations in colors between themes--at least for one background color.

Reply:
We may be able to accomplish some of these using the a pattern like we do for icons in the button slot, ie override token values that are used by the child: https://github.com/ni/nimble/blob/main/packages/nimble-components/src/patterns/button/styles.ts#L127

Reply:
If we have dedicated slots and guidelines about what is allowed to be slotted in them (action only button, start / icon ony icon, anchor only nimble-anchor, etc) then we can easily set those properties on those slots